### ISSUE TITLE
Doubt regarding `loss` while training.

### ISSUE THREAD

**Issue by vijaydwivedi75** [sglvladi/TensorFlowObjectDetectionTutorial#9]
I am training for Custom Object Detection using Mask RCNN in TensorFlow Object Detection. Therefore, I am to predict the object instance mask along with the bounding box.

pre-trained model : `mask_rcnn_inception_v2_coco`

Following is a snapshot of my training.

```INFO:tensorflow:global step 4178: loss = 0.0129 (2.796 sec/step)
INFO:tensorflow:global step 4179: loss = 0.8679 (1.942 sec/step)
INFO:tensorflow:global step 4179: loss = 0.8679 (1.942 sec/step)
INFO:tensorflow:global step 4180: loss = 0.0014 (2.833 sec/step)
INFO:tensorflow:global step 4180: loss = 0.0014 (2.833 sec/step)
INFO:tensorflow:global step 4181: loss = 0.0031 (3.290 sec/step)
INFO:tensorflow:global step 4181: loss = 0.0031 (3.290 sec/step)
INFO:tensorflow:global step 4182: loss = 0.0030 (2.745 sec/step)
INFO:tensorflow:global step 4182: loss = 0.0030 (2.745 sec/step)
INFO:tensorflow:global step 4183: loss = 1.0958 (2.748 sec/step 
```

**In this case, can you please tell me what is the `loss` here?**

**Comment by PHPirates** [comment]
I'm just a passer-by with not too much experience  in machine learning, but are you asking why there are such jumps in your loss? As that's the only strange thing I see. I have no idea what that means though, but it seems a general question about machine learning, not about this tutorial? Maybe someone else can help you here, but I have no idea, sorry.

I mean, when I take your question literally I can answer it, because the loss over there is 1.0958 at the end :)

**Comment by vijaydwivedi75** [comment]
My questions is not related to training loss and its variation w.r.t. the steps.

I am just unclear about **what is meant by this loss** while training a Mask RCNN? In a Mask RCNN, there are 3 parallel heads at the last layer,

1.  for detecting the class
2.  for predicting bounding box
3.  for predicting instance masks 

In such a case, **what is loss?**

**Comment by sglvladi** [comment]
@vijaydwivedi75 

Unfortunately I cannot provide you with a confident answer. Although this is a tutorial to assist you in installing and running some basic examples in Tensorflow, it is not intended as a discussion forum.

Thus, I would suggest that you post any queries you have on [StackOverflow](https://stackoverflow.com/questions/tagged/tensorflow), as advised by [Tensorflow](https://www.tensorflow.org/community/) themselves.

I hope you find an answer to your problem.
